Sources/TokamakDOM/App.swift

```diff
@@ -0,0 +1,140 @@
+// Copyright 2020 Tokamak contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//  Created by Carson Katri on 7/16/20.
+//
+
+import JavaScriptKit
+import OpenCombine
+import TokamakCore
+
+public typealias App = TokamakCore.App
+
+public typealias ScenePhase = TokamakCore.ScenePhase
+
+private enum ScenePhaseObserver {
+  static var publisher = CurrentValueSubject<ScenePhase, Never>(.active)
+
+  static func observe() {
```

Comment:
Another cue to look for here is `document.hasFocus()` — if it’s `false` but `document.visiblityState` is `visible`, the page is `.inactive`. I couldn’t find out if there were any events that trigger when this changes, though.

Reply:
Doesn't seem like it looking through MDN. If you find something lmk.